tools/@aws-cdk/eslint-plugin/lib/rules/no-invalid-path.ts

```diff
@@ -0,0 +1,94 @@
+import { Rule } from 'eslint';
+import { isProdFile } from '../private/is-prod-file';
+import * as path from 'path';
+import * as fs from 'fs';
+
+function isPathJoinFuncCall(node: any): boolean {
+  return node.callee?.property?.name === 'join';
+}
+
+function allArgumentsExpected(node: any): boolean {
+  // Outside of the first argument, all arguments should be strings
+  const components = node.arguments.slice(1);
+  return components.every((a: any) => a.value !== undefined);
+}
+
+function hasSlashes(args: string[]): boolean {
+  return args.some((a) => a.includes('/'));
+}
+
+function firstArgIsDirname(node: any): boolean {
+  return node.arguments[0].name && node.arguments[0].name === '__dirname';
+}
+
+function argumentList(node: any): string[] {
+  // Already confirmed that first argument is '__dirname', so can safely remove it
+  const args: string[] = node.arguments.slice(1).map((a: any) => { return a.value; });
+  return args;
+}
+
+function recreatePath(args: string[]): string {
+  return `path.join(__dirname, '${args.join('\', \'')}')`;
+}
+
+export function create(context: Rule.RuleContext): Rule.NodeListener {
+  return {
+    CallExpression(node: any) {
+      if (!isProdFile(context.getFilename())) {
+        return;
+      }
+
+      if (isPathJoinFuncCall(node)) {
+        if (node.arguments.length === 0) {
+          // ERROR: this is 'path.join()'
+          context.report({ node, message: '\'path.join()\' is not a valid path. You must specify arguments into the function.'})
+          return;
+        }
+
+        if (!allArgumentsExpected(node)) {
+          // ERROR: unexpected non-string in the argument list
+          context.report({ node, message: 'Part of the arguments to \'path.join()\' is not a string. Only the first argument can be \'__dirname\'.'})
+          return;
+        }
+
+        // We currently do not lint any path.join without '__dirname' as the first argument
+        if (!firstArgIsDirname(node)) {
+          return;
+        }
```

Comment:
Can discuss what we want to do in this thread

Reply:
This is perfect!